variables.tf

```
@@ -294,6 +306,18 @@ variable "terraformer_role_name" {
   default     = "Terraformer"
 }
 
+variable "valid_assessment_id_regex" {
+  type        = string
+  description = "A regular expression that specifies valid assessment identifiers (e.g. \"^ASMT[[:digit:]]{4}$\")."
+  default     = ""
```

We may as well validate that the string provided is a valid regex.  I think this should do just that:
```suggestion
  default     = ""
  validation {
    # regexall() fails if the second argument is not a valid regex
    condition     = can(regexall("", var.valid_assessment_id_regex))
    error_message = "The string provided must be a valid RE2 regular expression."
  }
```

---

It turns out that this is not necessary.  It looks like the `precondition` block is evaluated before this input validation condition.  When I tested your suggestion with some invalid regexes, I did not get the expected error message (`"The string provided must be a valid RE2 regular expression."`). Instead, I got errors like the following, which I think are sufficient:

```
│ Error: Invalid function argument
│ 
│   on assessment_validation.tf line 15, in resource "null_resource" "validate_assessment_id":
│   15:       condition     = length(regexall(var.valid_assessment_id_regex, var.assessment_id)) == 1
│     ├────────────────
│     │ var.valid_assessment_id_regex is "^foo\\"
│ 
│ Invalid value for "pattern" parameter: invalid regexp pattern: trailing backslash at end of expression in .
```

```
│ Error: Invalid function argument
│ 
│   on assessment_validation.tf line 15, in resource "null_resource" "validate_assessment_id":
│   15:       condition     = length(regexall(var.valid_assessment_id_regex, var.assessment_id)) == 1
│     ├────────────────
│     │ var.valid_assessment_id_regex is "^\\p$"
│ 
│ Invalid value for "pattern" parameter: invalid regexp pattern: invalid character class range in \p$.
```

```
│ Error: Invalid function argument
│ 
│   on assessment_validation.tf line 15, in resource "null_resource" "validate_assessment_id":
│   15:       condition     = length(regexall(var.valid_assessment_id_regex, var.assessment_id)) == 1
│     ├────────────────
│     │ var.valid_assessment_id_regex is "^RV[[:foo:]]{4}$"
│ 
│ Invalid value for "pattern" parameter: invalid regexp pattern: invalid character class range in [:foo:].
```

@jsf9k Are you ok with this level of validation?

---

Ah, OK.  Yep, you already have it covered.